compiler/src/dotty/tools/dotc/transform/CheckUnused.scala

```diff
@@ -252,31 +253,45 @@ class CheckUnused extends MiniPhase:
   private def traverseAnnotations(sym: Symbol)(using Context): Unit =
     sym.denot.annotations.foreach(annot => traverser.traverse(annot.tree))
 
+  private def aggregateUnused(data: UnusedData, res: UnusedData.UnusedResult)(using Context): Unit =
+    data.unusedAggregate match {
+      case None =>
+        data.unusedAggregate = Some(res)
+      case Some(prevUnused) =>
+        val intersection = res.warnings.filter(sym => prevUnused.warnings.contains(sym))
```

Comment:
After changing `UnusedData#warnings` to `Set`, change this line to just intersection.